### PR TITLE
Memory: Add bios memory

### PR DESCRIPTION
**Pull request by guerinoni**
Signed-off-by: Federico Guerinoni <guerinoni.federico@gmail.com>

### CODE REVIEW

**Comment by FedericoBruzzone** [comment]
Looks good. 
Let's talk about if it's better to merge #92 before this or before this one.

**Review by AlessioC31 — APPROVED** [review]
Looks good to me too.

Out of curiosity, do you know who should populate this memory region? Because I've read that GBA stores in BIOS some procedures (lz compression, division, etc). Should it be read from a bios file? Or all the `.gba` files populate it?

**Review by FedericoBruzzone — APPROVED** [review]

File: emu/src/internal_memory.rs [hunk 1]
```diff
@@ -3,6 +3,9 @@ use crate::io_device::IoDevice;
 use crate::io_registers::LCDRegisters;
 
 pub struct InternalMemory {
+    /// From 0x00000000 to 0x00003FFF (16 KBytes)
+    bios_system_rom: [u8; 0x3FFF],
```
Comment (gallottino) [inline emu/src/internal_memory.rs:7]: The size is wrong. The right one is `0x4000`. I open a issue about this #97.

File: emu/src/internal_memory.rs [hunk 2]
```diff
@@ -28,6 +31,7 @@ impl Default for InternalMemory {
 impl InternalMemory {
     pub const fn new() -> Self {
         Self {
+            bios_system_rom: [0; 0x3FFF],
```
Comment (gallottino) [inline emu/src/internal_memory.rs:34]: Same of above.

**Review by gallottino — CHANGES_REQUESTED** [review]

File: emu/src/internal_memory.rs (hunk 1 above)
Comment (guerinoni) [reply emu/src/internal_memory.rs]: Done :)

File: emu/src/internal_memory.rs (hunk 2 above)
Comment (guerinoni) [reply emu/src/internal_memory.rs]: Done :)

**Review by gallottino — APPROVED** [review]

**Review by AlessioC31 — APPROVED** [review]

**Pull request merged**